GVRf/Framework/framework/src/main/jni/objects/textures/base_texture.h

```diff
@@ -85,6 +85,12 @@ class BaseTexture: public Texture {
         }
     }
 
+    void updateFromBuffer(int internalFormat, int width, int height, int format, int type, void* data) {
+        glBindTexture(GL_TEXTURE_2D, getId());
+        glTexImage2D(GL_TEXTURE_2D, 0, internalFormat, width, height, 0, format, type, data);
```

Comment:
are we just updating texture here, is the memory preallocated? If we are updating texture data, then PBO would be best option and if not then glTexSubImage2D.